docs/standard/assembly/reference-assemblies.md

```diff
@@ -0,0 +1,74 @@
+
+If you want to distribute reference assemblies with NuGet packages, you must include them in the *ref\\* subdirectory under the package directory instead of in the *lib\\* subdirectory used for implementation assemblies.
+
+## Reference assemblies structure
+
+Reference assemblies are an expansion of the related concept, *metadata-only assemblies*. Metadata-only assemblies have their method bodies replaced with a single `throw null` body, but include all members except anonymous types. The reason for using `throw null` bodies (as opposed to no bodies) is so that PEVerify can run and pass (thus validating the completeness of the metadata).
+
+Reference assemblies further remove metadata (private members) from metadata-only assemblies:
+
+- A reference assembly only has references for what it needs in the API surface. The real assembly may have additional references related to specific implementations. For instance, the reference assembly for `class C { private void M() { dynamic d = 1; ... } }` does not reference any types required for `dynamic`.
+- Private function-members (methods, properties, and events) are removed in cases where their removal doesn't observably impact compilation. If there are no [InternalsVisibleTo](xref:System.Runtime.CompilerServices.InternalsVisibleToAttribute) attributes, internal function members are also removed.
+
+The metadata in reference assemblies continues to retain information about the following:
+
+- All types, including private and nested types.
+- All attributes, even internal ones.
+- All virtual methods.
+- Explicit interface implementations. 
+- Explicitly implemented properties and events, because their accessors are virtual.
+- All fields of structures. 
+
+Reference assemblies include an assembly-level [ReferenceAssembly](xref:System.Runtime.CompilerServices.ReferenceAssemblyAttribute) attribute. This attribute may be specified in source; then the compiler won't need to synthesize it. Because of this attribute, runtimes will refuse to load reference assemblies for execution (but they can still be loaded in reflection-only mode).
+
+Exact reference assembly structure details depend on the compiler version. Newer versions may choose to exclude more metadata if it is determined as not affecting the public API surface.
+
+> [!NOTE]
+> Information in this section is applicable only to reference assemblies generated by Roslyn compilers starting from C# version 7.1 or Visual Basic version 15.3. The structure of reference assemblies for .NET Framework versions released before that time can differ in some details. For example, they might have totally empty method bodies instead of the `throw null` body. But the general principle still applies: they don't have usable method implementations and contain metadata only for members that have an observable impact from a public API perspective.
```

Comment:
Oh I see. If the above refers to how Roslyn generates them, then that's probably correct. I was thinking about how generate reference assemblies in CoreFx.

Reply:
Edited to reflect that .NET Core is special case too. I included information about reference assemblies generated by Roslyn, because it was already documented in compiler options descriptions. How CoreFx generates reference assemblies, on the other hand, is largely implementation details and i think shouldn't be included.